### ISSUE TITLE
Vendoring breaks use of ecs-cli as dependency

### ISSUE THREAD

**Issue by shatil** [aws/amazon-ecs-cli#289]
Not sure what I'm doing wrong here, but I'm trying to convert `docker-compose.yml` files (to easily layer them using overrides) to ECS Task Definitions using [`utils.ConvertToTaskDefinition`](http://localhost:6060/pkg/github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose/#ConvertToTaskDefinition).

Attempting to `go run convert.go` using `go version go1.8.3 darwin/amd64`:

```go
package main

import (
	"fmt"

	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose"

	"github.com/docker/libcompose/config"
	"github.com/docker/libcompose/project"
)

func main() {
	taskDefinitionName := "my-task"
	taskRoleARN := "my-role-arn"
	context := &project.Context{
		ComposeFiles: []string{"docker-compose.yml"},
		ProjectName:  taskDefinitionName,
	}
	serviceConfigs := &config.ServiceConfigs{}
	taskDefinition, err := utils.ConvertToTaskDefinition(
		taskDefinitionName,
		context,
		serviceConfigs,
		taskRoleARN,
	)
	if err != nil {
		fmt.Println("Couldn't convert to Task Definition")
	}
	fmt.Print(taskDefinition)
}
```

Yields the following with exit code 2:

```
# command-line-arguments
./convert.go:25: cannot use context (type *"github.com/docker/libcompose/project".Context) as type *"github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/project".Context in argument to utils.ConvertToTaskDefinition
./convert.go:25: cannot use serviceConfigs (type *"github.com/docker/libcompose/config".ServiceConfigs) as type *"github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/config".ServiceConfigs in argument to utils.ConvertToTaskDefinition
```

I'm able to build `ecs-cli` fine following `README.md`'s instructions. I tried `git checkout v0.4.0` of `github.com/docker/libcompose` to no avail.

**Comment by PettitWesley** [comment]
@shatil Thank you for your question. Are you familiar with the [godep](https://github.com/tools/godep) project? It is what we use for vendoring. I replicated your project, and was able to get it to build by first running `godep save` in the project directory (which creates a vendor directory and fixes imports). 

Let me know if you have any questions.

**Comment by shatil** [comment]
Ah, I see. Since I was able to build `ecs-cli` without it... I ignored it.

`godep update` worked, but `godep save` in `github.com/aws/amazon-ecs-cli/ecs-cli` yielded `godep: Package (github.com/Azure/go-ansiterm) not found` with exit code 1. I have some research to go on `godep` :(

**Comment by PettitWesley** [comment]
@shatil, yeah unfortunately `godep` is confusing. Also, please take a look at the section of our ReadMe for "Contributing to the CLI", we include instructions for setting up a local environment for working with the CLI code. As you can see in the ReadMe, we recommend running `godep restore` in `github.com/aws/amazon-ecs-cli/ecs-cli` when you first pull down the project- this pulls all of our dependencies and adds them to your `GOPATH` (unfortunately in a detached HEAD state however). In my previous comment when I suggested that you run `godep save` in your project directory, I was assuming that you had created a separate go project with the ECS CLI as a dependency. 

Also, if you are merely interested in converting between different container formats, you could 
 also investigate [container-transform](https://github.com/micahhausler/container-transform). It advertises the ability to convert between different docker container formats, like ECS Task-Definition and Docker compose, as well as others. (Disclaimer: I am merely bringing this project to your attention, I can not vouch for its effectiveness, and I have never used it.)

Finally, if you have any suggestions on how to improve the ECS CLI, or find any bugs in our code, please do not hesitate, to post an issue on our Github page!

**Comment by shatil** [comment]
Thanks! I definitely have some more reading to do on `godep` usage, and must do a better job of following directions.

I also discovered the promising `container-transform` project, but encountered a bug early on: https://github.com/micahhausler/container-transform/issues/68

It proved simpler for my use case to write a small script to merge YAML files (which I learned was a feature thanks to `ecs-cli`) and convert the keys to ECS Task Definition, even where the values are differently organized, like `environment`.
